### PR TITLE
TIP-725 - Refactoring the attributes index screen with the new PEF ar…

### PR DESCRIPTION
**Pull request by Alban44**
…chitecture

[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**
Refactoring the attributes index screen with the new PEF architecture
Complete the PR #6052

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/config/requirejs.yml [hunk 1]
```diff
@@ -68,6 +68,22 @@ config:
...
+                    pim_catalog_price_collection:    money
+                    pim_catalog_multiselect:         tags
+                    pim_catalog_simpleselect:        tag
+                    pim_catalog_file:                file
+                    pim_catalog_image:               camera-retro
+                    pim_catalog_boolean:             check
+                    pim_catalog_date:                calendar
+                    pim_catalog_metric:              beaker
+                    pim_reference_data_multiselect:  tags
+                    pim_reference_data_simpleselect: tag
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/config/requirejs.yml]: unindent all this plz (4 chars for each level)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                this.config = config.config;
+
+                BaseForm.prototype.initialize.apply(this, arguments);
+            },
+
+            /**
+             * Create the dialog modal and bind clicks
+             */
+            createModal: function (attributeTypesMap) {
+
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js]: useless line

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                return this;
+            },
+
+            /**
+             * Format the map to an array and sort attributeTypes by label
+             * @param attributeTypesMap => Map of attributeTypes
+             * @returns {Array}
+             */
+            formatAndSortAttributeTypesByLabel: function (attributeTypesMap) {
+
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js]: same

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    if (attributeTypesMap.hasOwnProperty(key)) {
+                        sortedAttributeTypesByLabel.push({
+                            code: key,
+                            label: __('pim_enrich.entity.attribute_label.' + attributeTypesMap[key])
+                        });
+                    }
+                }
+
+                sortedAttributeTypesByLabel.sort(function (a, b) {
+                    return a.label > b.label ? 1 : -1;
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes/create-button.js]: you can use localeCompare to do this
http://stackoverflow.com/questions/2167602/optimum-way-to-compare-strings-in-javascript

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/tab/attributes/create-modal-content.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div id="attribute-modal-content">
+    <div class="AknButtonList AknButtonList--twoColumns">
+        <% _.each(attributeTypes, function(attributeType){ %>
+            <a class="AknButtonList-item AknButton AknButton--withIcon attribute-choice"
+               href="<%- generateRoute('pim_enrich_attribute_create', { attribute_type: attributeType.code }) %>">
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/tab/attributes/create-modal-content.html]: Please check with @juliensnz , I think we have to not introduce new `<a href>` elements

File: src/Pim/Bundle/EnrichBundle/Resources/views/Attribute/index.html.twig [hunk 6]
```diff
@@ -1,64 +1,30 @@
...
-                                modal.$el.find('.modal-footer').remove();
-
-                                modal.$el.on('click', 'a.attribute-choice', function(e) {
-                                    e.preventDefault();
-                                    modal.close();
-                                    modal.$el.remove();
-                                    navigation.getInstance().navigate('#url=' + $(this).attr('href'), { trigger: true });
-                                });
-                            }
+    <script>
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/views/Attribute/index.html.twig]: Please check if this block is compatible with new hashnav with @juliensnz @phaseinducer @tamarasaurus

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeExtension.php [hunk 7]
```diff
@@ -26,9 +21,8 @@ class AttributeExtension extends \Twig_Extension
      *
      * @param array $icons
      */
-    public function __construct(array $icons, IdentifiableObjectRepositoryInterface $attributeRepository)
+    public function __construct(IdentifiableObjectRepositoryInterface $attributeRepository)
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeExtension.php]: This is a BC break, you have to add it to the CHANGELOG

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/tab/attributes/create-modal-content.html (hunk 5 above)
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/form/tab/attributes/create-modal-content.html]: Yeah, never use href. Please catch the event properly and use a span :)

File: src/Pim/Bundle/EnrichBundle/Twig/AttributeExtension.php (hunk 7 above)
Comment (Alban44) [inline src/Pim/Bundle/EnrichBundle/Twig/AttributeExtension.php]: Done!

**Pull request merged**